crates/utils/src/utils.rs

```diff
@@ -165,29 +165,17 @@ pub fn get_ip(conn_info: &ConnectionInfo) -> IpAddr {
   )
 }
 
-pub fn clean_url_params(mut url: Url) -> Url {
+pub fn clean_url_params(url: &Url) -> Url {
+  let mut url_out = url.to_owned();
```

Comment:
You just moved the clone inside the function. In this case it doesnt make any difference as far as i see, but in general its better to take an owned value as param if thats what you need. Because then the caller can pass in an owned value if he has one available, and avoid an extra clone.

Reply:
Its mostly about the functional programming paradigm of not altering state when not necessary.  Its strange to be passing in something that you're altering, and then having it return that same altered thing. Best to just pass in refs to most functions, seems to be the most flexible.

Reply:
Hmm it is a bit more elegant in the call site, so in this case im okay with it.